File: Projeto-7_Kayron/src/funcionarioAdministrativo.py
from .usuarioAutenticavel import UsuarioAutenticavel
from .relatorio import Relatorio
from .tipoUsuario import TipoUsuario


class FuncionarioAdministrativo(UsuarioAutenticavel, Relatorio):
    def __init__(self, nome: str, senha: str, setor: str):
        super().__init__(nome, senha)
        self.tipo = TipoUsuario.FUNCIONARIO_ADMINISTRATIVO
        self.setor = setor

    def autenticar(self, senha: str) -> bool:
        return self._verificar_senha(senha)

    def gerar_relatorio(self):
        print(f"Relatório do Funcionário {self.nome}:")
        print(f"Setor: {self.setor}")

File: Projeto_7/src/funcionarioAdministrativo.cpp
#include "FuncionarioAdministrativo.h"
#include <iostream>

FuncionarioAdministrativo::FuncionarioAdministrativo(
    const std::string& nome,
    const std::string& email,
    const std::string& senha,
    const std::string& setor)
    : UsuarioAutenticavel(nome, email, TipoUsuario::FUNCIONARIO_ADMINISTRATIVO, senha),
      setor(setor) {}

bool FuncionarioAdministrativo::autenticar(const std::string& senhaInformada) const {
    return senha == senhaInformada;
}

void FuncionarioAdministrativo::gerarRelatorio() const {
    std::cout << "Funcionário: " << nome
              << " | Setor: " << setor << std::endl;
}